### PR TITLE
Womtool endpoint: indicate when a workflow itself is valid, not considering inputs

### LINKED ISSUE #4824: Womtool endpoint: separate field for input errors
- Problems with inputs should be related to the input name in a machine-readable way so that a UI can display the information next to the field it pertains to (Doug)
- If a WDL is valid and there is only a problem with inputs, indicate that somehow and still return I/O (DA)
- ~If a WDL is invalid, don't return the keys for I/O... because they can be seen as falsely indicating that the WDL has no inputs or outputs. (DMohs)~ moved into https://github.com/broadinstitute/cromwell/issues/4892

### PR DESCRIPTION
**Pull request by aednichols**
Closes #4824 

I didn't do
>a sort of input_errors map with input names as keys and error(s) as values

because
1. The underlying WOM creation returns pre-formatted strings (e.g. `Required workflow input 'wf_hello.hello.addressee' not specified`) and changing that interface would be a huge undertaking
2. There's no neat way to handle extraneous inputs, since they obviously would not match any input name key in the map

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,10 @@ backend {
 }
 ```
 
+### Womtool endpoint
+
+When submitting a workflow together with an inputs JSON to the `/describe` endpoint, the response now differentiates between an invalid workflow and a valid workflow with invalid inputs. Look for the `inputErrors` key in the response or see the Swagger for more information.
```
Comment (kshakir) [inline CHANGELOG.md]: > Look for the `inputErrors` key in the response

ToL: Is `inputErrors` always present now? So presence of the *key* does not automatically mean there exists an error? The returned list has to be non-empty, right?

File: services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala [hunk 2]
```diff
@@ -153,20 +153,20 @@ class WomtoolServiceInCromwellActorSpec extends ServicesSpec("Womtool") {
...
     }
 
-    "return invalid for a valid workflow with the wrong inputs" in {
+    "return inputErrors for a valid workflow with the wrong inputs" in {
 
       val wsfc = wsfcConjurer(workflowSource = Option(TestData.wdlValid), inputsJson = TestData.bogusInputs)
 
       check(DescribeRequest(wsfc), DescribeSuccess(
-        description = WorkflowDescription(valid = false, errors = List("Required workflow input 'wf_hello.hello.addressee' not specified"))))
+        description = WorkflowDescription(valid = true, inputErrors = List("Required workflow input 'wf_hello.hello.addressee' not specified"), name = "wf_hello", inputs = List(InputDescription("wf_hello.hello.addressee", WomStringType, "String", false, None)), submittedDescriptorType = Map("descriptorType" -> "WDL", "descriptorTypeVersion" -> "1.0"))))
```
Comment (cjllanwarne) [inline services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala]: Oh interesting. I'm... unsure... whether I'm cool with `valid = true` coexisting with `inputErrors`...

**Comment by cjllanwarne** [comment]
Might or might not block release. We'll await Adam's verdict on Monday morning

File: CHANGELOG.md (hunk 1 above)
Comment (aednichols) [reply CHANGELOG.md]: Yes. For better or for worse we are doing "all keys always present". I know this is a source of perennial debate and if there is a consensus to change we can.

File: services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala (hunk 2 above)
Comment (aednichols) [reply services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala]: Hmm, this sounds like a good suggestion. I'll give it a try.

**Comment by aednichols** [comment]
@cjllanwarne suggestion implemented

File: services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala [hunk 3]
```diff
@@ -195,15 +197,15 @@ class WomtoolServiceInCromwellActorSpec extends ServicesSpec("Womtool") {
...
     }
 
-    "return invalid for a valid no-inputs workflow with extraneous inputs" in {
+    "return valid = false, validWorkflow = true for a valid no-inputs workflow with extraneous inputs" in {
 
       val wsfc = wsfcConjurer(workflowSource = Option(TestData.wdlValidNoInputs), inputsJson = TestData.bogusInputs)
 
       check(DescribeRequest(wsfc), DescribeSuccess(
-        description = WorkflowDescription(valid = false, errors = List("WARNING: Unexpected input provided: foo.bar"))))
+        description = WorkflowDescription(valid = false, errors = List("WARNING: Unexpected input provided: foo.bar"), validWorkflow = true, name = "wf_hello", inputs = List.empty, submittedDescriptorType = Map("descriptorType" -> "WDL", "descriptorTypeVersion" -> "1.0"))))
```
Comment (cjllanwarne) [inline services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala]: I thought these were being put into a separate `inputErrors` field (per the PR title?)

Or are we now saying "if you see `validWorkflow` but still have errors, all errors indicate input errors"?

Comment (aednichols) [reply services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala]: Good catch, the PR title is out of date.

**Pull request merged**